File: backend/TrenchAPI/Core/Entities/Messung.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TrenchAPI.Core.Entities
{
    public class Messung : EntityObject
    {
        [Required]
        public int MesseinstellungID { get; set; }

        [ForeignKey(nameof(MesseinstellungID))]
        public virtual Messeinstellung Messeinstellung { get; set; }

        [Column(TypeName = "timestamp with time zone")]
        public DateTime Anfangszeitpunkt { get; set; }

        [Column(TypeName = "timestamp with time zone")]
        public DateTime Endzeitpunkt { get; set; }

        [Column(TypeName = "varchar")]
        public string Name { get; set; } = "";

        [Column(TypeName = "decimal")]
        public decimal Tauchkernstellung { get; set; }

        [Column(TypeName = "decimal")]
        public decimal Pruefspannung { get; set; }

        [Column(TypeName = "varchar")]
        public string Notiz { get; set; } = "";
    }
}

File: frontend/src/app/configuration/messung/interfaces/messung.ts
import { MeasurementSetting } from "../../measurement-settings/interfaces/measurement-settings";

export interface Messung {
    id: number | null,
    name: string | null,
    messeinstellung: MeasurementSetting | null,
    messeinstellungId: number | null,
    anfangszeitpunkt: Date | null,
    endzeitpunkt: Date | null,
    tauchkernstellung: number | null,
    pruefspannung: number | null,
    notiz: string | null,
}
